File: tests/output/vanilla-js/model/restaurant/paiement-item.ts
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

import {DO_DATE_HEURE, DO_ID, DO_PRIX} from "../../domains";

export interface PaiementItem {
    factureId: number;
    swileCardId: number;
    dateCommande: string;
    montantTotal: number;
}

export const PaiementItemEntity = {
    factureId: {
        type: "field",
        name: "factureId",
        domain: DO_ID,
        isRequired: true,
        label: "FactureId"
    },
    swileCardId: {
        type: "field",
        name: "swileCardId",
        domain: DO_ID,
        isRequired: true,
        label: "SwileCardId"
    },
    dateCommande: {
        type: "field",
        name: "dateCommande",
        domain: DO_DATE_HEURE,
        isRequired: true,
        label: "DateCommande"
    },
    montantTotal: {
        type: "field",
        name: "montantTotal",
        domain: DO_PRIX,
        isRequired: true,
        label: "Montant total"
    }
} as const;

File: tests/output/jpa/src/main/javagen/restaurant/jpa_jdbc_resttemplate/dtos/restaurant/PaiementItem.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package restaurant.jpa_jdbc_resttemplate.dtos.restaurant;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import org.springframework.data.relational.core.mapping.Column;

import jakarta.annotation.Generated;
import jakarta.validation.constraints.NotNull;

/**
 * Détail d'un paiement.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public record PaiementItem(
	@Column("id") @NotNull Integer factureId,
	@Column("swi_id") @NotNull Integer swileCardId,
	@Column("com_date_commande") @NotNull LocalDateTime dateCommande,
	@Column("com_montant_total") @NotNull BigDecimal montantTotal
) {
}
